tests/testthat/test-forecaster-utils.R

```diff
@@ -15,70 +15,40 @@ test_that("sanitize_args_predictors_trainer", {
 })
 
 test_that("id generation works", {
-  simple_ex <- list(dplyr::tribble(
-    ~forecaster, ~trainer, ~ahead, ~pop_scaling, ~lags,
-    "scaled_pop", "linreg", 1, TRUE, c(1, 2)
-  ), dplyr::tribble(
-    ~forecaster, ~ahead, ~pop_scaling, ~trainer, ~lags,
-    "scaled_pop", 1, TRUE, "linreg", c(1, 2)
-  ))
+  # Same arguments but scrambled.
+  simple_ex <- list(
+    dplyr::tribble(
+      ~forecaster, ~trainer, ~pop_scaling, ~lags,
+      "scaled_pop", "linreg", TRUE, c(1, 2)
+    ), dplyr::tribble(
+      ~forecaster, ~pop_scaling, ~trainer, ~lags,
+      "scaled_pop", TRUE, "linreg", c(1, 2)
+    ), dplyr::tribble(
+      ~trainer, ~forecaster, ~pop_scaling,
+      "linreg", "scaled_pop", TRUE,
+    ), dplyr::tribble(
+      ~trainer, ~pop_scaling, ~forecaster,
+      "linreg", TRUE, "scaled_pop",
+    )
+  )
   same_ids <- map(simple_ex, add_id)
-  # the order shouldn't matter for the names
   expect_equal(same_ids[[1]]$id, same_ids[[2]]$id)
-  # the list version and the
-  list_ahead_ex <- list(forecaster = "scaled_pop", trainer = "linreg", ahead = 1, pop_scaling = TRUE, lags = c(1, 2))
-  expect_equal(single_id(list_ahead_ex), same_ids[[1]]$id)
-  # order shouldn't matter for the list either
-  list_ahead_scrambled <- list(lags = c(1, 2), trainer = "linreg", ahead = 1, pop_scaling = TRUE, forecaster = "scaled_pop")
-  expect_equal(single_id(list_ahead_scrambled), single_id(list_ahead_ex))
-  # the list version returns the same whether ahead is included
-  list_ex <- list(forecaster = "scaled_pop", trainer = "linreg", pop_scaling = TRUE, lags = c(1, 2))
-  expect_equal(single_id(list_ahead_ex), single_id(list_ex, 1))
```

Comment:
why did you drop these? I guess `get_single_id` has a different format than `single_id`?

Reply:
I think partly because I dropped re-ordering from get_single_id and moved it to add_id and things were actually different here. But also these lists were basically the same as what was in the tibble above (minus the ahead), so I just added them to the tibble, so get_single_id is implicitly covered above.

Reply:
tbh testing `get_single_id` is probably worth more than `add_id`. Both run basically immediately, which is why I had these fairly redundant tests.

Reply:
the code coverage is pretty much the same calling `add_id`, but i added a couple direct `get_single_id` calls